src/common/apollo.ts

```diff
@@ -143,14 +143,12 @@ const makeInstanceMiddleware = (opts: ApolloClientOpts) => {
    *
    * If identifier is set directly, use that, or fall back to request hostname.
    */
-  const { instanceHostname, instanceIdentifier } = opts;
+  const { instanceHostname, instanceIdentifier, locale } = opts;
```

Comment:
I wonder why it wasn't done like this before 🤔 is `opts.locale` defined in all cases?

Reply:
I'm not entirely sure, it's pretty difficult to track. At least here (https://github.com/kausaltech/kausal-paths-ui/blob/main/src/pages/_app.tsx#L377)  it is marked with a exclamation mark and set at least to `'en'` earlier ( https://github.com/kausaltech/kausal-paths-ui/blob/main/src/pages/_app.tsx#L296 ) but I wouldn't be surprised if I'm missing something. Also I couldn't get into trouble when testing with Zuerich, Espoo and Longmont.

But I actually implemented this first by giving the locale as context to every query and stashed that implementation, so switching to use that instead would be quick. I'm just worried that the locale will be forgotten from future queries.

Reply:
https://github.com/kausaltech/kausal-paths-ui/pull/122#discussion_r1785711697

> I would've done it the same! I don't quite understand the "Use locale directive in all GraphQL queries to make sure translated strings are always fetched" part though, I don't see any code related to that except removing it from the context parameter.
> 
> Or are we just now using the pre-existing `@locale` directive [here](https://github.com/kausaltech/kausal-paths-ui/blob/main/src/common/apollo.ts)?

Yeah, the `@locale` directive existed already, now just instead of getting the locale from the query context, we trust that it is found in the `opts`. The rest of the code was there already, the found locale is then taken into use here: https://github.com/kausaltech/kausal-paths-ui/blob/b2243242fa16e21697f408472e9cf89a278c8aa5/src/common/apollo.ts#L157

Buuut I get that it's a maybe a bit iffy and difficult to follow, there's surely room for improvement. I'm still okay with sticking with the query context implementation if you think that's better, or if you have some other suggestions!

Reply:
I have not used the test environment yet, how do I deploy there? :stuck_out_tongue: Should I merge to somewhere else than `main`?

Reply:
I figured it out now, tested a couple of different instances with different language configurations and couldn't break it. I'll resolve this now and go on with the PR :)